packages/circuit-ui/components/ImageInput/ImageInput.tsx

```diff
@@ -44,14 +52,34 @@ export interface ImageInputProps
    * An accessible label for the "clear" icon button.
    */
   clearButtonLabel: string;
+  /**
+   * A callback function to call when the user has selected an image.
+   */
+  onChange: (event: File | null) => Promise<void>;
+  /**
+   * A callback function to call when the input is cleared.
+   */
+  onClear: () => void;
+  /**
+   * ...
+   */
+  validationHint?: string;
+  /**
+   * ...
+   */
+  invalid?: boolean;
+  /**
+   * ...
+   */
+  loadingLabel: string;
 }
 
 const HiddenInput = styled.input(
-  ({ theme }) => css`
+  ({ theme, invalid }: StyleProps & { invalid: boolean }) => css`
     ${hideVisually()};
 
     &:focus + label {
-      ${focusOutline({ theme })};
+      ${!invalid && focusOutline({ theme })};
```

Comment:
Why is the focus outline hidden when the input is invalid? Do they overlap/clash somehow?

Reply:
It did clash, but now I think it makes more sense to keep the focus styles, at least until we have specific styles for the focused AvatarInput. At least there's a visual cue that way. We can't really use the same styles as the normal form elements since AvatarInput borders need to be thicker to be visible – I'll talk with designers to improve this as a follow-up

Reply:
Done as part of the larger styles cleanup/refactor in 48fa172